media/src/Settings.svelte

```diff
@@ -110,31 +109,41 @@
     const togglePolls = () => (isPollsOpen = !isPollsOpen);
 
     const uploadScene = (e) => {
-            const reader = new FileReader();
-            reader.onload = (evt) => {
-                let upload = [];
-                try {
-                    upload = JSON.parse(evt.target.result);
-                } catch (SyntaxError) {
-                    alert('Improper JSON formatting');
-                }
-                if (upload.length < 33) {
-                    for (let i = 0; i < upload.length; i++) {
-                        objects = makeObject(
-                            null,
-                            upload[i].kind,
-                            upload[i].params,
-                            objects,
-                            socket
-                        );
-                    }
-                } else {
-                    alert('Object limit of 32 per upload.');
+        const reader = new FileReader();
+        reader.onload = (evt) => {
+            let upload = [];
+            try {
+                upload = JSON.parse(evt.target.result);
+            } catch (SyntaxError) {
+                alert('Improper JSON formatting');
+            }
+            if (upload.length < 33) {
+                for (let i = 0; i < upload.length; i++) {
+                    objects = makeObject(
+                        null,
+                        upload[i].kind,
+                        upload[i].params,
+                        objects,
+                        socket
+                    );
                 }
+            } else {
+                alert('Object limit of 32 per upload.');
+            }
 
-            };
-            reader.readAsText(e.target.files[0], 'utf-8');
-        }
+        };
+        reader.readAsText(e.target.files[0], 'utf-8');
+    }
+
+    const downloadScene = () => {
+        let json = JSON.stringify(objects);
+        const blob = new Blob([json], {type: "application/json",});
+        let a = document.createElement('a');
+        a.download = new Date().toISOString().split('T')[0] + '.json';
```

Comment:
I've tested and this looks good. Just a minor suggestion, let's prepend a descriptive string to the filename so it's obvious what the file is, and maybe break this out into its own function as it gets more complex. e.g.:
```
const makeFilename = () => {
    const formattedDate = new Date().toISOString().split('T')[0];
    return '3demos-scene-' + formattedDate + '.json';
};

const downloadScene = () => {
    // ...
    a.download = makeFilename();
```

Reply:
Changes made :)